streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -589,4 +608,204 @@ boolean sendingOldValueEnabled() {
+                                                             final Serde<V> thisValueSerde,
+                                                             final Serde<KO> otherKeySerde,
+                                                             final Serde<VR> joinedValueSerde) {
+
+        ((KTableImpl<?, ?, ?>) other).enableSendingOldValues();
+        enableSendingOldValues();
+
+        final String repartitionerName = builder.newProcessorName(REPARTITION_NAME);
+        final String repartitionTopicName = JOINOTHER_NAME + repartitionerName + "-TOPIC";
+        final String repartitionProcessorName = repartitionerName + "-" + SELECT_NAME;
+        final String repartitionSourceName = repartitionerName + "-SOURCE";
+        final String repartitionSinkName = repartitionerName + "-SINK";
+        final String joinOneToOneName = repartitionerName + "-TABLE";
+
+        // repartition original => intermediate topic
+        final KTableRepartitionerProcessorSupplier<K, KO, V> repartitionProcessor =
+                new KTableRepartitionerProcessorSupplier<>(keyExtractor);
+
+        final CombinedKeySerde<KO, K> combinedKeySerde = new CombinedKeySerde<>(otherKeySerde, thisKeySerde);
+
+        //Create the partitioner that will partition CombinedKey on just the foreign portion (right) of the combinedKey.
+        final CombinedKeyByForeignKeyPartitioner<KO, K, V> partitioner;
+        if (null == foreignKeyPartitioner)
+            partitioner = new CombinedKeyByForeignKeyPartitioner<>(combinedKeySerde, repartitionTopicName);
+        else
+            partitioner = new CombinedKeyByForeignKeyPartitioner<>(combinedKeySerde, repartitionTopicName, foreignKeyPartitioner);
+
+        //The processor for this table. It does two main things:
+        // 1) Loads the data into a stateStore, to be accessed by the KTableKTablePrefixJoin processor.
+        // 2) Drives the join logic from this.
+        //    Returns the data keyed on K. Discards the CombinedKey as it is no longer needed after this stage.
+        final String joinByPrefixName = builder.newProcessorName(BY_PREFIX);
+
+        final String thisStateStoreName = builder.newStoreName(STATE_STORE_NAME);
+        final ForeignKeySingleLookupProcessorSupplier<K, KO, V, VO, V> joinOneToOne =
+                new ForeignKeySingleLookupProcessorSupplier(thisStateStoreName, ((KTableImpl<?, ?, ?>) other).valueGetterSupplier(), joiner);
+
+        final KeyValueBytesStoreSupplier thisRocksDBRef = new RocksDbKeyValueBytesStoreSupplier(thisStateStoreName);
+        final StateStore prefixScannableDBRef = thisRocksDBRef.get();
+        final Materialized foreignMaterialized = Materialized.<CombinedKey<KO, K>, V, KeyValueStore<Bytes, byte[]>>as(prefixScannableDBRef.name())
+                //Need all values to be immediately available in the rocksDB store.
+                //No easy way to flush cache prior to prefixScan, so caching is disabled on this store.
+                .withCachingDisabled()
```

Comment:
I notice that in 2.x that I may be able to rework this to allow for enabled cache using a `prefixScan` function similar to `ThreadCache.range`. I will have to look into this a bit more, though I don't think it will affect performance much since I anticipate RocksDB prefixScan to take the longest overall.

Reply:
Might be, its one of the places I got stuck once. From experience I can tell that its working sufficiently well w/o cache. I think rocks does a pretty good job in not seeking around to randomly on the disk

Reply:
I'll leave it out for now. If someone else thinks otherwise, they can speak up or it can be done in a subsequent PR.